src/h5web/metadata-viewer/utils.ts

```diff
@@ -1,7 +1,45 @@
+import {
+  HDF5BaseType,
+  HDF5Endianness,
+  HDF5TypeClass,
+} from '../providers/hdf5-models';
+
 export function renderShapeDims(dims: number[]): string {
   if (dims.length > 1) {
     return `${dims.join(' x ')} = ${dims.reduce((acc, value) => acc * value)}`;
   }
 
   return dims.toString();
 }
+
+export function renderEndianness(endianness: HDF5Endianness): string {
+  switch (endianness) {
+    case 'LE':
+      return 'Little-endian';
+
+    case 'BE':
+      return 'Big-endian';
+
+    case 'Native':
+      return 'Native';
+
+    case 'Not applicable':
+      return '';
+
+    default:
+      return '';
+  }
+}
+
+export function renderBaseType(type: HDF5BaseType): string {
+  if (type.class === HDF5TypeClass.String) {
+    const { length, charSet } = type;
+    return `${charSet} string (size:
+      ${Number.isInteger(length) ? length : 'variable'})`;
+  }
+
+  const { endianness, size } = type;
+  return `${renderEndianness(endianness)}
+  ${type.class === HDF5TypeClass.Integer && type.unsigned ? 'unsigned' : ''}
+  ${size}-bit ${type.class.slice(4).toLowerCase()}`;
```

Comment:
If you agree with the single-row idea, I think it would make sense to show the most relevant information first rather than to make a grammatically correct phrase. For instance:

- "Float, 32-bit, big-endian"
- "Integer, unsigned 16 bit, little-endian"
- "String, variable length"
- "String, length=42

Not sure about the exact formatting/construction. Perhaps Silx can be of some inspiration here.